domain/user/state/state.go

```diff
@@ -291,6 +291,27 @@ func (st *State) SetPasswordHash(ctx context.Context, uuid user.UUID, passwordHa
 	})
 }
 
+// BootstrapAddUserWithPassword adds a new user to the database during bootstrap with the
+// provided password hash and salt. If the user already exists an error that
+// satisfies usererrors.AlreadyExists will be returned. if the creator does
+// not exist that satisfies usererrors.UserCreatorUUIDNotFound will be returned.
+func BootstrapAddUserWithPassword(
```

Comment:
This logic is identical to the code we have in `AddUserWithPassword` can we just make one function that can be used for both cases?

Reply:
done